### DISCUSSION TITLE
customize cc_librarary

### DISCUSSION THREAD

**Discussion by maneeshsinghxavo** [bazelbuild/bazel#18733] [Q&A]
Hi,
as per example 
https://github.com/bazelbuild/examples/tree/main/configurations/cc_binary_selectable_copts

I would like to customize the cc_libarry 
I wrote sample implementation 

```
def _library_transition_rule_impl(ctx):
    actual_library = ctx.attr.actual_library[0]
    outfile = ctx.actions.declare_file("lib" + ctx.label.name +".a")
    cc_library_outfile = actual_library[DefaultInfo].files.to_list()[0] # take .a file extension

    #this contains .a or .so extension
    print("outfile.path is {}".format(outfile.path))
    print("cc_library_outfile.path is {}".format(cc_library_outfile.path))

    ctx.actions.run_shell(
        inputs = [cc_library_outfile],
        outputs = [outfile],
        command = "cp %s %s" % (cc_library_outfile.path, outfile.path),
    )
    return [
        DefaultInfo(
              #executable = outfile,
              default_runfiles = actual_library[DefaultInfo].default_runfiles,
              files = depset([outfile])
        ),
         OutputGroupInfo(all_files = depset([outfile])),
    ]
```

sample output 

- ERROR: /home/manesingh/work/DAI/minerva/examples-1/cpp-tutorial/stage3/main/BUILD:15:10: in deps attribute of cc_binary rule //main:hello-world_native_binary: library_transition_rule rule '//main:hello-greet' is misplaced here (expected cc_library, objc_library, cc_proto_library or cc_import) and '//main:hello-greet' does not have mandatory providers: 'CcInfo'. Since this rule was created by the macro 'cc_binary', the error might have been caused by the macro implementation

- ERROR: /home/manesingh/work/DAI/minerva/examples-1/cpp-tutorial/stage3/main/BUILD:15:10: in deps attribute of cc_binary rule //main:hello-world_native_binary: library_transition_rule rule '//lib:hello-time' is misplaced here (expected cc_library, objc_library, cc_proto_library or cc_import) and '//lib:hello-time' does not have mandatory providers: 'CcInfo'. Since this rule was created by the macro 'cc_binary', the error might have been caused by the macro implementation

- ERROR: /home/manesingh/work/DAI/minerva/examples-1/cpp-tutorial/stage3/main/BUILD:15:10: in cc_binary rule //main:hello-world_native_binary: target '//lib:hello-time' is not visible from target '//main:hello-world_native_binary'. Check the visibility declaration of the former target if you think the dependency is legitimate

- ERROR: /home/manesingh/work/DAI/minerva/examples-1/cpp-tutorial/stage3/main/BUILD:15:10: Analysis of target '//main:hello-world_native_binary' failed

- ERROR: Analysis of target '//main:hello-world' failed; build aborted: 
- INFO: Elapsed time: 0.215s
- INFO: 0 processes.
- FAILED: Build did NOT complete successfully (38 packages loaded, 165 targets configured)

Could someone help to figure this out?

Thanks

**Comment by oquenchil** [comment]
Sorry not exactly clear what you want to do, would you mind explaining it in a bit more detail and breaking down the problem?
